src/Select/SelectInput.js

```diff
@@ -136,15 +136,11 @@ class SelectInput extends React.Component<AllProps, State> {
         value = child.props.value;
       }
 
-      this.props.onChange(
-        {
-          ...event,
-          target: {
-            value,
-          },
-        },
-        child,
-      );
+      event.persist();
+      // $FlowFixMe
+      event.target = { ...event.target, value };
+      // $FlowFixMe
+      this.props.onChange(event, child);
```

Comment:
`$FlowFixMe` should not be needed, let's solve the problem.  We should not introduce workarounds into a hardened codebase, I don't want to accept PRs with `$FlowFixMe` unless it is a last resort, which I am unconvinced it is in this case.